File: 백준/Bronze/2884. 알람 시계/알람 시계.js
function solution(input) {
    let hour = input[0];
    let minute = input[1] - 45;
    if(minute < 0) {
        if(hour === 0) {
            hour += 23;
        }
        else {
            hour -= 1;
        }
        minute = 60 + minute;
    }
    console.log(hour, minute);
}

const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,    
});

let input;
let list = [];

rl.on("line", function (line) {
    input = line;
    rl.close();
}).on("close", function () {
    list = input.split(' ').map((el) => Number(el));

    solution(list);
    process.exit();
});

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
"""
1. 입력: H, M
2. 현재 시간 - 45분
"""
# H, M 받아오기
H, M = map(int, input().split())

# M이 45보다 작은 경우 -> H = H-1 / M = 60 - (45 - M)
if M < 45:
    M = M + 60 - 45

    if H > 0:
        H = H - 1
    else:
        H = 23
else:
    M = M - 45

print(H, M)